homeassistant/helpers/config_entry_oauth2_flow.py

```diff
@@ -77,6 +79,12 @@ async def async_resolve_external_data(self, external_data: Any) -> dict:
     async def async_refresh_token(self, token: dict) -> dict:
         """Refresh a token and update expires info."""
         new_token = await self._async_refresh_token(token)
+        # Force int for non-compliant oauth2 providers
+        try:
+            new_token["expires_in"] = int(new_token["expires_in"])
+        except ValueError as err:
+            _LOGGER.warning("Error converting expires_in to int: %s", err)
+            return token
```

Comment:
This will cause issues down the line. We should probably not even catch this error here.

Reply:
👌